loader_hub/github_repo/base.py

```diff
@@ -123,30 +148,33 @@ def _check_filter_directories(self, tree_obj_path: str) -> bool:
 
         :return: True if the tree object should be allowed, False otherwise
         """
+        if self._filter_directories is None:
+            return True
         filter_directories, filter_type = self._filter_directories
         print_if_verbose(
             self._verbose,
             f"Checking {tree_obj_path} whether to {filter_type} it"
             + f" based on the filter directories: {filter_directories}",
         )
 
-        if filter_type == self.FilterType.EXCLUDE:
-            return not any(
-                tree_obj_path.startswith(directory)
-                or directory.startswith(tree_obj_path)
-                for directory in filter_directories
-            )
-        elif filter_type == self.FilterType.INCLUDE:
-            return any(
-                tree_obj_path.startswith(directory)
-                or directory.startswith(tree_obj_path)
-                for directory in filter_directories
-            )
-        else:
-            raise ValueError(
-                f"Unknown filter type: {filter_type}. "
-                "Please use either 'ignore' or 'include'."
-            )
+        match filter_type:
```

Comment:
this would break backwards compatibility with older python versions (only supported on python 3.10), let's keep as if/else instead

Reply:
> sorry @ahmetkca for the long delay, thanks for putting out this PR! one comment around still supporting older python versions. also does the unit test pass for you?

It does but for GitHub client test you need to set `GITHUB_TOKEN` env variable which I honestly don't like but it makes network requests to check the test case.

Reply:
> this would break backwards compatibility with older python versions (only supported on python 3.10), let's keep as if/else instead
btw What is the oldest python version LlamaIndex supports? so I can test it on that version which should cover higher versions.